backend/api/v1/rollout_service.go

```diff
@@ -193,6 +200,107 @@ func (s *RolloutService) ListRollouts(ctx context.Context, req *connect.Request[
+
+	getFilter = func(expr celast.Expr) (string, error) {
+		switch expr.Kind() {
+		case celast.CallKind:
+			functionName := expr.AsCall().FunctionName()
+			switch functionName {
+			case celoperators.LogicalAnd:
+				return getSubConditionFromExpr(expr, getFilter, "AND")
+			case celoperators.Equals:
+				variable, value := getVariableAndValueFromExpr(expr)
+				switch variable {
+				case "task_type":
+					taskType, ok := value.(string)
+					if !ok {
+						return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("task_type value must be a string"))
+					}
+					// Validate task type
+					if _, ok := v1pb.Task_Type_value[taskType]; !ok {
+						return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("invalid task_type value: %s", taskType))
+					}
+					// Convert v1pb.Task_Type to storepb.Task_Type
+					v1TaskType := v1pb.Task_Type(v1pb.Task_Type_value[taskType])
+					storeTaskType := convertToStoreTaskType(v1TaskType)
+					// Query tasks that have the specified type
+					positionalArgs = append(positionalArgs, storeTaskType.String())
+					return fmt.Sprintf("EXISTS (SELECT 1 FROM task WHERE task.pipeline_id = pipeline.id AND task.type = $%d)", len(positionalArgs)), nil
+				default:
+					return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("unsupported variable %q", variable))
+				}
+			case celoperators.In:
+				variable, value := getVariableAndValueFromExpr(expr)
+				switch variable {
+				case "task_type":
+					rawList, ok := value.([]any)
+					if !ok {
+						return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("invalid list value %q for %v", value, variable))
+					}
+					if len(rawList) == 0 {
+						return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("empty list value for filter %v", variable))
+					}
+					var placeholders []string
+					for _, raw := range rawList {
+						taskType, ok := raw.(string)
+						if !ok {
+							return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("task_type value must be a string"))
+						}
+						// Validate task type
+						if _, ok := v1pb.Task_Type_value[taskType]; !ok {
+							return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("invalid task_type value: %s", taskType))
+						}
+						// Convert v1pb.Task_Type to storepb.Task_Type
+						v1TaskType := v1pb.Task_Type(v1pb.Task_Type_value[taskType])
+						storeTaskType := convertToStoreTaskType(v1TaskType)
+						positionalArgs = append(positionalArgs, storeTaskType.String())
+						placeholders = append(placeholders, fmt.Sprintf("$%d", len(positionalArgs)))
+					}
+					// Query tasks that have any of the specified types
+					return fmt.Sprintf("EXISTS (SELECT 1 FROM task WHERE task.pipeline_id = pipeline.id AND task.type IN (%s))", strings.Join(placeholders, ", ")), nil
+				default:
+					return "", connect.NewError(connect.CodeInvalidArgument, errors.Errorf("unsupported variable %q", variable))
```

Comment:
Can we add `TaskTypes` to `PipelineFind`?  Should we concat SQL filter strings here?

Reply:
Let's keep it consistent with other resources, and make unified modifications if needed.